### PR TITLE
Feature/spring integration

### PR DESCRIPTION
**Pull request by 8r1m2t0n3**

### CODE REVIEW

File: src/main/java/org/andersen/homework/MainClass.java [hunk 1]
```diff
@@ -86,85 +83,11 @@ public static void usersStand() {
...
-
-    for (User user : userService.getAll()) {
-      userService.delete(user.getId());
-    }
-    for (Ticket t : ticketService.getAll()) {
-      ticketService.delete(t.getId());
-    }
-  }
-
   public static void main(String[] args) {
```
Comment (Darya-Drutsko) [inline src/main/java/org/andersen/homework/MainClass.java]: put this method on top of the class
In current way it's hard to read what happened

File: src/main/java/org/andersen/homework/MainClass.java [hunk 2]
```diff
@@ -13,19 +10,19 @@
...
 import org.andersen.homework.service.UserService;
+import org.andersen.homework.util.TicketGenerator;
+import org.springframework.context.ApplicationContext;
+import org.springframework.context.annotation.AnnotationConfigApplicationContext;
 
 public class MainClass {
 
-  private static final String TICKET_AND_USER_TABLES_CREATION_SQL_FILE = "db/sql/ticket_and_user_tables_creation.sql";
-  private static final DBConnectionSingleton DB_CONNECTION = DBConnectionSingleton.getInstance();
+  private static void ticketsStand(ApplicationContext context) {
```
Comment (Darya-Drutsko) [inline src/main/java/org/andersen/homework/MainClass.java]: what does ticketStand do? It's not obvious from the name

File: src/main/java/org/andersen/homework/config/ApplicationContextConfig.java [hunk 3]
```diff
@@ -0,0 +1,52 @@
...
+    dataSource.setUser(dbUser);
+    dataSource.setPassword(dbPassword);
+    dataSource.setURL(dbUrl);
+    return dataSource;
+  }
+
+  @Bean
+  public boolean initializeDatabase() {
+    new ResourceDatabasePopulator(new ClassPathResource(TICKET_AND_USER_TABLES_CREATION_SQL_FILE)).execute(dataSource());
+    return Boolean.TRUE; // dummy value
```
Comment (Darya-Drutsko) [inline src/main/java/org/andersen/homework/config/ApplicationContextConfig.java:45]: why do we need this?

File: src/main/java/org/andersen/homework/service/UserService.java [hunk 4]
```diff
@@ -2,22 +2,30 @@
...
-import org.andersen.homework.model.dao.jdbc.JdbcDaoFactory;
+import lombok.RequiredArgsConstructor;
 import org.andersen.homework.model.dao.jdbc.UserDaoJdbc;
 import org.andersen.homework.model.entity.user.User;
+import org.springframework.stereotype.Service;
+import org.springframework.transaction.annotation.Transactional;
 
+@Service
+@RequiredArgsConstructor
+@Transactional(readOnly = true)
```
Comment (Darya-Drutsko) [inline src/main/java/org/andersen/homework/service/UserService.java:13]: What does do this annotation on class level?

File: src/main/java/org/andersen/homework/MainClass.java (hunk 2 above)
Comment (8r1m2t0n3) [reply src/main/java/org/andersen/homework/MainClass.java]: it's just an example of all implemented, related to `Ticket`, logic usage, going to rename it better way

File: src/main/java/org/andersen/homework/config/ApplicationContextConfig.java (hunk 3 above)
Comment (8r1m2t0n3) [reply src/main/java/org/andersen/homework/config/ApplicationContextConfig.java:45]: every bean must return a value, but this bean just populates db and doesn't need to return any value. This dummy value needs just to satisfy the framework requirements

File: src/main/java/org/andersen/homework/service/UserService.java (hunk 4 above)
Comment (8r1m2t0n3) [reply src/main/java/org/andersen/homework/service/UserService.java:13]: Applying `@Transactional(readOnly = true)` at the class level ensures that all methods in the class are treated as read-only by default, means that these methods are optimized for read operations and cannot modify db. For methods that should modify db I added `@Transactional` annotation to its signature to enables write operations for them while keeping the rest of the class read-only.

File: src/main/java/org/andersen/homework/config/ApplicationContextConfig.java (hunk 3 above)
Comment (Darya-Drutsko) [reply src/main/java/org/andersen/homework/config/ApplicationContextConfig.java:45]: why don't return ResourceDatabasePopulator?

File: src/main/java/org/andersen/homework/MainClass.java (hunk 2 above)
Comment (Darya-Drutsko) [reply src/main/java/org/andersen/homework/MainClass.java]: rename please

File: src/main/java/org/andersen/homework/config/ApplicationContextConfig.java (hunk 3 above)
Comment (8r1m2t0n3) [reply src/main/java/org/andersen/homework/config/ApplicationContextConfig.java:45]: `new ResourceDatabasePopulator(new ClassPathResource(TICKET_AND_USER_TABLES_CREATION_SQL_FILE)).execute(dataSource())` returns no value, so i can't return it, let it be boolean as return value for this bean

Comment (8r1m2t0n3) [reply src/main/java/org/andersen/homework/config/ApplicationContextConfig.java:45]: this bean is useless after initialization, so it doesn't matter what will be returned

**Pull request merged**